pymatgen/util/typing.py

```diff
@@ -5,30 +5,33 @@
 
 from __future__ import annotations
 
+from collections.abc import Sequence
 from os import PathLike as OsPathLike
 from typing import TYPE_CHECKING, Any, Union
 
 from pymatgen.core import Composition, DummySpecies, Element, Species
 
 if TYPE_CHECKING:  # needed to avoid circular imports
+    from typing_extensions import TypeAlias
+
     from pymatgen.analysis.cost import CostEntry  # type: ignore[attr-defined]
     from pymatgen.analysis.phase_diagram import GrandPotPDEntry, PDEntry, TransformedPDEntry
     from pymatgen.entries import Entry
     from pymatgen.entries.computed_entries import ComputedEntry, ComputedStructureEntry, GibbsComputedStructureEntry
     from pymatgen.entries.exp_entries import ExpEntry
 
 
-PathLike = Union[str, OsPathLike]
-PbcLike = tuple[bool, bool, bool]
+PathLike: TypeAlias = Union[str, OsPathLike]
```

Comment:
[docs](https://docs.python.org/3/library/typing.html#typing.TypeAlias) recommend not to use `TypeAlias`:

> Deprecated since version 3.12: [TypeAlias](https://docs.python.org/3/library/typing.html#typing.TypeAlias) is deprecated in favor of the [type](https://docs.python.org/3/reference/simple_stmts.html#type) statement

Reply:
Thanks! I would fix this!

Reply:
I noticed `TypeAlias` is deprecated after Python 3.12 (which we haven't touched yet).

And `type` is new for Python 3.12 too? Maybe remove this type annotation altogether?